android/src/main/java/com/tailscale/ipn/App.kt

```diff
@@ -119,15 +124,26 @@ class App : UninitializedApp(), libtailscale.AppContext, ViewModelStoreOwner {
     viewModelStore.clear()
   }
 
+  @Volatile
   private var isInitialized = false
 
+  private fun ensureInitialized() {
+    if (!isInitialized) {
+        initOnce() // Calls the synchronized initialization logic
+    }
+}
+
   @Synchronized
   private fun initOnce() {
     if (isInitialized) {
       return
     }
     isInitialized = true
```

Comment:
The method is synchronized so it's OK, but let's defer setting isInitialized to the end of the method since, technically, nothing isInitialzied at this point.

Does kotlin have a built in "run this block of code once and only once"  construct?  In Swift, we would do dispatch_once() here (or use a closure) instead of a volatile bool and the synchronization.

Reply:
it doesn't afaik. I think the best we can do is the combination of the volatile isInitialized and the synchronization.